addons/loyalty/models/loyalty.py

```diff
@@ -0,0 +1,96 @@
+    rule_domain = fields.Char()
+
+
+class LoyaltyReward(models.Model):
+    _name = 'loyalty.reward'
+    _description = 'Loyalty Reward'
+
+    name = fields.Char(index=True, required=True, help='An internal identification for this loyalty reward')
+    loyalty_program_id = fields.Many2one('loyalty.program', string='Loyalty Program', help='The Loyalty Program this reward belongs to')
+    minimum_points = fields.Float(help='The minimum amount of points the customer must have to qualify for this reward')
+    reward_type = fields.Selection([('gift', 'Free Product'), ('discount', 'Discount')], required=True, help='The type of the reward', default="gift")
+    gift_product_id = fields.Many2one('product.product', string='Gift Product', help='The product given as a reward')
+    point_cost = fields.Float(string='Reward Cost', help="If the reward is a gift, that's the cost of the gift in points. If the reward type is a discount that's the cost in point per currency (e.g. 1 point per $)")
+    discount_product_id = fields.Many2one('product.product', string='Discount Product', help='The product used to apply discounts')
+    discount_type = fields.Selection([
+        ('percentage', 'Percentage'),
+        ('fixed_amount', 'Fixed Amount')], default="percentage",
+        help="Percentage - Entered percentage discount will be provided\n" +
+            "Amount - Entered fixed amount discount will be provided")
+    discount_percentage = fields.Float(string="Discount", default=10,
+            help='The discount in percentage, between 1 and 100')
+    discount_apply_on = fields.Selection([
+        ('on_order', 'On Order'),
+        ('cheapest_product', 'On Cheapest Product'),
+        ('specific_products', 'On Specific Products')], default="on_order",
+        help="On Order - Discount on whole order\n" +
+            "Cheapest product - Discount on cheapest product of the order\n" +
+            "Specific products - Discount on selected specific products")
+    discount_specific_product_ids = fields.Many2many('product.product', string="Products",
+            help="Products that will be discounted if the discount is applied on specific products")
+    discount_max_amount = fields.Float(default=0,
+            help="Maximum amount of discount that can be provided")
+    discount_fixed_amount = fields.Float(string="Fixed Amount", help='The discount in fixed amount')
+    minimum_amount = fields.Float(string="Minimum Order Amount")
+
+    @api.constrains('reward_type', 'gift_product_id')
+    def _check_gift_product(self):
+        if self.filtered(lambda reward: reward.reward_type == 'gift' and not reward.gift_product_id):
+            raise ValidationError(_('The gift product field is mandatory for gift rewards'))
+
+    @api.model
+    def create(self, vals):
+        reward = super().create(vals)
+        if reward.reward_type == 'discount' and not vals.get('discount_product_id', False):
```

Comment:
`dict.get` returns the falsy value `None` by default
```suggestion
        if reward.reward_type == 'discount' and not vals.get('discount_product_id'):
```